sellotape/sellotape_dj/main_app/util/youtube_func.py

```diff
@@ -0,0 +1,60 @@
+from django.utils import timezone
+from django.shortcuts import get_object_or_404
+from ..models import Stream, Profile
+# pip install --upgrade google-api-python-client
+from apiclient.discovery import build
+
+api_key = "AIzaSyCAL9HUYTpY9vX05h2Wnk5CAmYmPvSxX8E"
+
+# get - username.
+# extracts each video in user's youtube playlist
+# insert the extracted data into Streams DB
+# return - number of videos that were inserted, or 404/403 with error message.
+
+
+def get_channel_videos(username):
+    # get youtube_link of username if exist.
+    profile = get_object_or_404(Profile, user__username=username)
+    youtube_link = profile.youtube_link
+    if not youtube_link:
+        return 404, 'there is no youtube_link related to this user'

```

Comment:
I think that it shouldn't return a 404 (or any error status code), assuming for a caller method to handle that...

Those checks are important and good that you have them 👍   But maybe it should just exit the function and print the reason.